src/main/java/org/terasology/manualLabor/systems/ShearingSystem.java

```diff
@@ -0,0 +1,122 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.manualLabor.systems;
+
+import org.joml.Vector3f;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.engine.core.Time;
+import org.terasology.engine.entitySystem.entity.EntityBuilder;
+import org.terasology.engine.entitySystem.entity.EntityManager;
+import org.terasology.engine.entitySystem.entity.EntityRef;
+import org.terasology.engine.entitySystem.event.ReceiveEvent;
+import org.terasology.engine.entitySystem.prefab.Prefab;
+import org.terasology.engine.entitySystem.systems.BaseComponentSystem;
+import org.terasology.engine.entitySystem.systems.RegisterSystem;
+import org.terasology.engine.logic.characters.events.AttackEvent;
+import org.terasology.engine.logic.delay.DelayManager;
+import org.terasology.engine.logic.delay.PeriodicActionTriggeredEvent;
+import org.terasology.engine.logic.inventory.events.DropItemEvent;
+import org.terasology.engine.logic.location.LocationComponent;
+import org.terasology.engine.registry.In;
+import org.terasology.engine.rendering.assets.material.Material;
+import org.terasology.engine.rendering.assets.skeletalmesh.SkeletalMesh;
+import org.terasology.engine.rendering.logic.SkeletalMeshComponent;
+import org.terasology.gestalt.assets.ResourceUrn;
+import org.terasology.gestalt.assets.management.AssetManager;
+import org.terasology.manualLabor.components.ShearableComponent;
+
+import java.util.Optional;
+
+/**
+ * This system is responsible for handling the logic for shearing. A sheep shearing cycle consists of a shearing switching to the model for
+ * sheared state, dropping an item in event of shearing and switching back to the non sheared state after a certain amount of time.
+ */
+@RegisterSystem
+public class ShearingSystem extends BaseComponentSystem {
+    public static final String SHEARING_ITEM = "ManualLabor:CrudeShears";
+    public static final int HAIR_REGROWTH_TIME = 3 * 60 * 1000; // 3 minutes in ms
+    public static final String HAIR_REGROWTH_ACTION_ID = "ManualLabor:HairRegrowthAction";
+    public static final String SHEARED_SHEEP_MESH = "WildAnimals:shearedSheep";
+    public static final String SHEARED_SHEEP_MATERIAL = "WildAnimals:shearedSheepSkin";
+    public static final String SHEEP_MESH = "WildAnimals:sheep";
+    public static final String SHEEP_MATERIAL = "WildAnimals:sheepSkin";
+    private static final Logger logger = LoggerFactory.getLogger(ShearingSystem.class);
+
+    @In
+    protected Time time;
+
+    @In
+    private EntityManager entityManager;
+
+    @In
+    private AssetManager assetManager;
+
+    @In
+    private DelayManager delayManager;
+
+    /**
+     * Executes changes occurring in event of shearing.
+     *
+     * @param entityRef Entity being sheared
+     */
+    @ReceiveEvent(components = {ShearableComponent.class})
+    public void onAttack(AttackEvent event, EntityRef entityRef) {
```

Comment:
I have some reservations about this part of the system. 
This very much feels like oh the entity is being attacked and has a shearable component, let's just remove the hair and drop the wool.
A much better approach could be :
The entity is being attacked and has a shearable component => the entity needs to be sheared => remove hair and drop wool
This in my opinion could be achieved in 2 ways -
1) With an event - After seeing that the entity being attacked has a shearable component and the cause was a shear, we send out an event to the entity like a ShearSheepEvent and have this functionality in the event handler for that
2) Without an event - simply have a separate function in the same file as in shearSheep(entity) and then the hair can be removed and wool can be dropped in this separate function. This would make it a lot cleaner and easier for the reader to understand.

@jdrueckert @keturn  thoughts?

Reply:
I'm not 100% sure this detour is necessary - although I understand where you're coming from. If we do it, then we should definitely do it with an event.